src/NServiceBus.Core/Persistence/Development/SagaPersister/DevelopmentSagaIdGenerator.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus.Features
+{
+    using System;
+    using Extensibility;
+    using NServiceBus.Sagas;
+
+    class DevelopmentSagaIdGenerator : ISagaIdGenerator
+    {
+        public Guid Generate(string propertyName, object propertyValue, SagaMetadata metadata, ContextBag context)
+        {
+            //here we assume single sagas since v6 doesn't allow more than one corr prop
+            // note that we still have to use a GUID since moving to a string id will have to wait since its a breaking change
+            return DeterministicGuid.Create(propertyValue);
```

Comment:
@Particular/ravendb-persistence-maintainers were ignored 😉 😭 

> are custom finders something we want to move down to persisters?

Does this mean that each persistence can have then a different API for finders?